### PR TITLE
DEV: Refactor composer-/topic-presence-display

### PR DESCRIPTION
**Pull request by CvX**
A followup to f05b984208e312ead39ce0ded560350d0b427ab2

* modifiers to keep track of components' lifecycles, instead of did-insert/did-update/willDestroy
* proper glimmer-friendly tracking in related models
* caching
* `@outletArgs`
* gjs

### CODE REVIEW

File: plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.gjs [hunk 1]
```diff
@@ -0,0 +1,149 @@
+import Component from "@glimmer/component";
+import { cached, tracked } from "@glimmer/tracking";
+import { service } from "@ember/service";
+import { modifier } from "ember-modifier";
+import { gt } from "truth-helpers";
+import UserLink from "discourse/components/user-link";
+import avatar from "discourse/helpers/avatar";
+import i18n from "discourse-common/helpers/i18n";
+
+export default class ComposerPresenceDisplay extends Component {
```
Comment (ZogStriP) [inline plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.gjs:10]: 😍

File: plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.gjs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      return;
+    }
+
+    const editChannel = this.presence.getChannel(
+      `/discourse-presence/edit/${this.args.model.post.id}`
+    );
+    editChannel.subscribe();
+    this.editChannel = editChannel;
+
+    return () => editChannel?.unsubscribe();
```
Comment (ZogStriP) [inline plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.gjs:65]: Why does this one use the `?.` operator when the other two don't?

File: plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.gjs [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    return users
+      .filter((u) => u.id !== this.currentUser.id)
+      .slice(0, this.siteSettings.presence_max_users_shown);
+  }
+
+  <template>
+    <div
+      {{this.setupReplyChannel}}
```
Comment (ZogStriP) [inline plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.gjs:116]: Thanks, I wasn't sure how to write "custom modifiers". Now I understand 👍

File: plugins/discourse-presence/assets/javascripts/discourse/components/topic-presence-display.gjs [hunk 4]
```diff
@@ -0,0 +1,77 @@
+import Component from "@glimmer/component";
+import { cached, tracked } from "@glimmer/tracking";
+import { service } from "@ember/service";
+import { modifier } from "ember-modifier";
+import { gt } from "truth-helpers";
+import UserLink from "discourse/components/user-link";
+import avatar from "discourse/helpers/avatar";
+import i18n from "discourse-common/helpers/i18n";
+
+export default class TopicPresenceDisplay extends Component {
```
Comment (ZogStriP) [inline plugins/discourse-presence/assets/javascripts/discourse/components/topic-presence-display.gjs:10]: 😍

**Review by ZogStriP — APPROVED** [review]
The code is so much betterer 👏

File: plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.gjs (hunk 2 above)
Comment (CvX) [reply plugins/discourse-presence/assets/javascripts/discourse/components/composer-presence-display.gjs:65]: right, it isn't needed anymore. removing!

**Pull request merged**